src/js/client/common-event-bindings.js

```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+/**
+ * GPII Binder Markup Events Grade
+ * This grade allows binding typical HTML events such as
+ * mouse clicks and keypress events to selectors each time
+ * the markup is rendered for a component. (Meaning they will
+ * also apply if a component is rerendered after a model refresh
+ * or similar situation)
+ *
+ * It adds a new area to a grades options called `markupEventBindings`
+ * which allows binding `selectors` to jQuery events. Theoretically
+ * other event constructs could be supported in the future, but only
+ * jQuery events are implemented at the time of writing.
+ *
+ * Example usage of adding a click handler to a selector productListLinks.
+ * ```
+ * markupEventBindings: {
+ *     productListLinks: {
+ *         // type: jQuery <- Defaults to jQuery but could be configured ITF
+ *         method: "click",
+ *         args: ["{that}.selectProduct"]
+ *     }
+ * }
+ * ```
+ */
+fluid.defaults("gpii.binder.bindMarkupEvents", {
+    mergePolicy: {
+        decorators: "noexpand"
+    },
+    events: {
+        onDomBind: null,
+        onDomUnbind: null
+    },
+    listeners: {
+        onMarkupRendered: "{that}.events.onDomBind.fire({that}, {that}.container)",
+        onDestroy: "{that}.events.onDomUnbind.fire({that}, {that}.container)",
```

Comment:
1. As mentioned above, please namespace these listeners;
2. Shall this "onDestroy" listener has a `priority: "first"`. It won't make sense to unbind after `{that}` has been destroyed.

Reply:
this is tidied up